unit_tests/parse_map_to_double_array.cc

```diff
@@ -82,6 +82,36 @@ TEST_CASE("Utilities::parse_map_to_double_array")
   "TestField"),
   {0.0,100.0,200.0,300.0,400.0,500.0});
 
+  {
+    INFO("check 9: ");
+    auto subentry_structure = std::make_shared<std::vector<unsigned int>>();
+
+    compare_vectors_approx(aspect::Utilities::parse_map_to_double_array ("C1:100, C2:200|100, C3:300, C4:400, C5:500, background:0",
+    {"C1","C2","C3","C4","C5"},
+    true,
+    "TestField",
+    true,
+    subentry_structure),
+    {0.0,100.0,200.0,100.0,300.0,400.0,500.0});
+
+    REQUIRE(*subentry_structure == std::vector<unsigned int>({1,1,2,1,1,1}));
+  }
+
+  {
+    INFO("check 10: ");
+    auto subentry_structure = std::make_shared<std::vector<unsigned int>>();
+
+    compare_vectors_approx(aspect::Utilities::parse_map_to_double_array ("C1:100, C2:200|, C3:300, C4:400, C5:500",
```

Comment:
Oh, we actually allow to use the | but then not have an entry? Is that intended?

Reply:
That is allowed by deal.II (it depends on the implementation of the `split_string_list` function, which explicitly allows this case). I think this is fine, likely nobody wants to use it.